### ISSUE TITLE
Discord rate limits the update frequency of the presence

### ISSUE THREAD

**Issue by evynhh** [tnychn/mpv-discord#21]
After a little bit, Discord starts to ratelimit the `AV:` line, plus it's mostly useless because of the "xx:xx left" line (which isn't rate limited due to it being set once) anyway. 
This also means that it won't receive events for pausing once rate limited, causing it to keep counting down even after pausing.
![image](https://user-images.githubusercontent.com/46719718/120058319-d8831600-bffe-11eb-9bdf-f32a6cced687.png)

**Comment by Reinbowsaur** [comment]
As 1.5 does not have this issue, you can downgrade in the meantime if its bothering enough.

**Comment by Kruchji** [comment]
Is there any update on this issue?

**Comment by tnychn** [comment]
The video progress (the `AV: 00:01:19/00:51:08` row) in Rich Presence requires frequent updates to keep it real time.
But I don't think there is a workaround to bypass Discord's rate limit.
Perhaps a solution to this issue is to use something else rather than video progress for the text.

Any suggestions on what the text could be?

**Comment by Kruchji** [comment]
Would it be possible to add an option (maybe in the config) to update the video progress less frequently? 

e.g. every 10 seconds, so Discord's rate limit wouldn't come into effect

**Comment by tnychn** [comment]
Yes, that would be a great solution. I will try to implement it when I'm free.

**Comment by Kruchji** [comment]
Maybe the rate needs to be even lower?

But also games like League of Legends show exact time of match so there probably is another way of doing this

**Comment by zeratax** [comment]
yeah that's what I was thinking about too. just deprecate that line and instead of updating based on an interval, updating after an action (pause,play,change title, etc) is performed

**Comment by mfederczuk** [comment]
Throwing my hat into the ring here to say that I was even blocked by Cloudflare after around 2 hours:

<img src="https://github.com/tnychn/mpv-discord/assets/14075939/dd2eb23c-ceb2-4e49-ba64-cde8004051bb" width="500" />

My only guess is that Cloudflare thinks that it was an attempted DoS due to the amount of calls sent to the servers.  
Now, while it's definitely possible that this _didn't_ result from this integration, I've never got something like this in all my years of using Discord (even with spamming the Discord servers myself a bit) and it only appeared after starting to use this integration.

I honestly think that this is f\*cking hilarious, but this is definitely a problem lmao

**Comment by Reinbowsaur** [comment]
> Throwing my hat into the ring here to say that I was even blocked by Cloudflare after around 2 hours:
> 
> <img alt="" width="500" src="https://user-images.githubusercontent.com/14075939/271546253-dd2eb23c-ceb2-4e49-ba64-cde8004051bb.png">
> My only guess is that Cloudflare thinks that it was an attempted DoS due to the amount of calls sent to the servers. Now, while it's definitely possible that this _didn't_ result from this integration, I've never got something like this in all my years of using Discord (even with spamming the Discord servers myself a bit) and it only appeared after starting to use this integration.
> 
> I honestly think that this is f*cking hilarious, but this is definitely a problem lmao

This was a global problem just over an hour ago, unrelated to this tool 👍

**Comment by mfederczuk** [comment]
> This was a global problem just over an hour ago, unrelated to this tool 👍

Ah, then I apologize. 🙇🏻  
Unfortunate timing for me to start using this.
